qiskit/dagcircuit/collect_blocks.py

```diff
@@ -235,6 +292,33 @@ def run(self, block):
         return blocks
 
 
+def split_block_into_layers(block):
+    """Splits a block of nodes into sub-blocks of non-overlapping instructions
+    (or, in other words, into depth-1 sub-blocks).
+    """
+    bit_depths = {}
+    layers = []
+
+    for node in block:
+        cur_bits = set()
+        cur_bits.update(node.qargs)
+        cur_bits.update(node.cargs)
+
+        cond = getattr(node.op, "condition", None)
+        if cond is not None:
+            cur_bits.update(condition_bits(cond))
```

Comment:
Does this work if the operation is not a `ControlFlowOp`?

Reply:
Oh, good catch! I did not realize that the function `condition_bits` was replaced by `condition_resources`. Fixed in [550b12d](https://github.com/Qiskit/qiskit-terra/pull/10254/commits/550b12df0593c26a7ce61cdef10bc0e8531295d2).

Reply:
I have somehow completely misinterpreted the question when I first read it. Which nodes get collected into blocks depends on the precise filtering function used. So one can collect blocks of non-conditional linear or Clifford gates, or, conversely, blocks of conditional gates. The latter was fixed in #9823. So somehow I thought that the question was if collecting blocks with `ControlFlowOps` works correctly, and it seems that it does, though I have no experimented too much on what happens on circuits with nested control flow operations. Though it's probably best to leave this investigation to after the release.

Reply:
Oh, and this is currently not recursive, i.e. it would not collect blocks of gates inside a control-flow block.

Reply:
Nah, I worded the question poorly, I was jumping between lots of reviews the past couple of days and what was clear in my head didn't come out clearly. But, you got what I was asking here anyway thanks.

As for handling control flow ops, having the block collection not handle it is fine. I think the path to fixing that would be to add support to `CollectAndCollapse' to recurse into blocks which shouldn't be too hard, but we can do that in a separate PR.